packages/sdk-browser-keysigner/index.ts

```diff
@@ -0,0 +1,77 @@
+import * as crypto from 'crypto'
+import { CredentialSigner, KeyAssertion } from '@dfns/sdk'
+import { toBase64Url } from '@dfns/sdk/utils'
+
+export class BrowserKeySigner implements CredentialSigner<KeyAssertion> {
+  constructor(
+    private options: {
+      privateKey: CryptoKey
+      credId: string
+      appOrigin: string
+      crossOrigin?: boolean
+      algorithm?: string
+    }
+  ) { }
+
+
+  minimizeBigInt = (value: Uint8Array): Uint8Array => {
+    const minValue = [0, ...value]
+    for (let i = 0; i < minValue.length; ++i) {
+      if (minValue[i] === 0) {
+        continue
+      }
+      if (minValue[i] > 0x7f) {
+        return new Uint8Array(minValue.slice(i - 1))
+      }
+      return new Uint8Array(minValue.slice(i))
+    }
+    return new Uint8Array([0])
+  }
+
+  rawSignatureToAns1 = (rawSignature: Uint8Array): Uint8Array => {
+    const r = rawSignature.slice(0, 32)
+    const s = rawSignature.slice(32)
+
+    const minR = this.minimizeBigInt(r)
+    const minS = this.minimizeBigInt(s)
+
+    return new Uint8Array([
+      0x30,
+      minR.length + minS.length + 4,
+      0x02,
+      minR.length,
+      ...minR,
+      0x02,
+      minS.length,
+      ...minS
+    ])
+  }
+
+
+  async sign(challenge: string): Promise<KeyAssertion> {
+    const clientData = JSON.stringify({
+      type: 'key.get',
+      challenge,
+      origin: this.options.appOrigin,
+      crossOrigin: this.options.crossOrigin ?? false,
+    })
+
+
+    const rawSignature = await crypto.subtle.sign(
+      { name: 'ECDSA', hash: { name: 'SHA-256' } },
```

Comment:
is `ECDSA` the only thing can be supported?

Reply:
Yes more protocols can be support, more configuration options will be needed though

Reply:
we already have the `algorithm` parameter that's not being used here. maybe we can change it's type from `string` to

```
type CryptoSubtleAlgorithm = Parameters<typeof crypto.subtle.sign>[0]
```

```
  constructor(
    private options: {
      privateKey: CryptoKey
      credId: string
      appOrigin: string
      crossOrigin?: boolean
      algorithm: CryptoSubtleAlgorithm
    }
  ) { }
```

Reply:
EdDSA isn't broadly supported yet, I'm going to remove `algorithm` from the params as we can figure it out from the key (`this.options.privateKey.algorithm.name`) and we have limited hash/curve support.

Reply:
what about RSA? did we say we want to encourage RSA because people get super confused about ECDSA for action signing and ECDSA in mpc?

Reply:
I'll think on that one, RSA can also be a bit confusing, ECDSA I guess does have different curves + hashing algorithms though we would only expect 1.